velox/dwio/parquet/tests/TestReader.cpp

```diff
@@ -208,3 +226,28 @@ TEST_F(ParquetReaderTest, testReadSampleEqualFilter) {
       {rangeVector<int64_t>(1, 16), rangeVector<double>(1, 16)});
   assertReadExpected(*rowReader, expected);
 }
+
+TEST_F(ParquetReaderTest, testDateRead) {
+  // date.parquet holds a single column (date: DATE) and
+  // 20 rows.
+  // Data is in plain uncompressed format:
+  //   date: [1970-01-01 .. 1970-01-20]
+  const std::string sample(getExampleFilePath("date.parquet"));
+
+  ReaderOptions readerOptions;
+  ParquetReader reader(
+      std::make_unique<FileInputStream>(sample), readerOptions);
+
+  EXPECT_EQ(reader.numberOfRows(), 20ULL);
+
+  auto type = reader.typeWithId();
+  EXPECT_EQ(type->size(), 1ULL);
+  auto col0 = type->childAt(0);
+  EXPECT_EQ(col0->type->kind(), TypeKind::DATE);
+
+  RowReaderOptions rowReaderOpts = getDateReaderOpts();
+  auto rowReader = reader.createRowReader(rowReaderOpts);
+
+  auto expected = vectorMaker_->rowVector({rangeVector<Date>(20, 0)});
```

Comment:
Maybe add another test case for Dates with negative integers (so those mapping to before epoch 1970-1-1)

Reply:
Good point. I added rows with such dates to the test data file.